### PR TITLE
Auto-generate katacoda docs

### PR DESCRIPTION
**Pull request by samj1912**
Signed-off-by: Sambhav Kothari <skothari44@bloomberg.net>

This commit adds a simple tool to auto-generate katacoda docs from our existing docs so that they don't diverge. It also adds a checker to github actions to ensure that they are up-to-date.

The auto-generator uses go's template library and uses a special form of html comment tags as delimiters so that the katacoda docs can co-exist with hugo docs without interfering with hugo's output.

As an example - I also ported over the app-journey scenario to show what it will look like.

Preview of what this will look like - https://www.katacoda.com/samj1912

After all the katacoda docs are migrated over, we can archive https://github.com/buildpacks/katacoda-scenarios

### CODE REVIEW

File: content/docs/app-developer-guide/run-an-app.md [hunk 1]
```diff
@@ -69,58 +74,75 @@ The `launcher` will source any profile scripts (for `non-direct` processes) and
...
 You can even override the buildpack-defined process types:
 
-```bash
+```
 docker run --rm --entrypoint launcher -it multi-process-app bash
+```
+<!--+- "{{execute interrupt}}"+-->
+
+Now let's exit this shell and run an alternative entrypoint - 
+```exit
```
Comment (jkutner) [inline content/docs/app-developer-guide/run-an-app.md:119]: is this right? I just looks a little odd to me

File: content/docs/app-developer-guide/using-cache-image.md [hunk 2]
```diff
@@ -20,58 +23,66 @@ The `--cache-images` flag must be specified in conjunction with the `--publish`
...
-Next we trust the `cnbs/sample-builder:bionic` builder
- in order to allow access to docker credentials when publishing.
 ```
 pack config trusted-builders add cnbs/sample-builder:bionic
 ```
+<!--+- "{{execute}}"+-->
 
-To build the `index.docker.io/buildpack-examples/cache-image-example` application image
- and the `index.docker.io/buildpack-examples/maven-cache-image:latest` cache image
+To build an the `localhost:5000/buildpack-examples/cache-image-example` application image
```
Comment (jkutner) [inline content/docs/app-developer-guide/using-cache-image.md:39]: Looks off. Remove "the"?

File: katacoda/scenarios/app-developer-guide/build-an-app.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+### 3. Run it
+
+```
+docker run --rm -p 8080:8080 sample-app
+```{{execute}}
+
+**Congratulations!**
+
+
+Now hit open the port "8080" of your host in your favorite browser and take a minute to enjoy the view.
```
Comment (jkutner) [inline katacoda/scenarios/app-developer-guide/build-an-app.md:49]: This reads a little odd. Maybe "Now open your favorite browser and point it to port 8080 of your host..."

File: content/docs/app-developer-guide/run-an-app.md (hunk 1 above)
Comment (samj1912) [reply content/docs/app-developer-guide/run-an-app.md:119]: Yeah - you need to exit the docker shell to proceed with the next step.

Comment (jkutner) [reply content/docs/app-developer-guide/run-an-app.md:119]: i didn't know ` ```exit` was legit

**Review by jkutner — APPROVED** [review]

File: content/docs/app-developer-guide/run-an-app.md (hunk 1 above)
Comment (samj1912) [reply content/docs/app-developer-guide/run-an-app.md:119]: Ah! Now I see what you meant! Yes - that is incorrect - good catch!

File: content/docs/app-developer-guide/build-an-app.md [hunk 4]
```diff
@@ -22,9 +26,10 @@ app.
 
 When using `pack`, you can run `pack builder suggest` for a list of suggested builders.
 
-```bash
+```
```
Comment (jromero) [inline content/docs/app-developer-guide/build-an-app.md:29]: For my personal learnings, is there a reason why this was removed? Is it incompatible, a preference, or should we be using `shell`?

**Review by jromero — APPROVED** [review]

**Comment by jromero** [comment]
LGMT! This is great!! 🎉

File: content/docs/app-developer-guide/build-an-app.md (hunk 4 above)
Comment (samj1912) [reply content/docs/app-developer-guide/build-an-app.md:29]: katacoda previously didn't support this. I tested it again and it seems to be working now. Will revert these changes.

Comment (samj1912) [reply content/docs/app-developer-guide/build-an-app.md:29]: Will fix this in a subsequent PR

**Pull request merged**